### PR TITLE
Remove unused NPM dependencies

### PR DESCRIPTION
**Pull request by robertbrignull**
I ran [depcheck](https://www.npmjs.com/package/depcheck) and then manually went through the results looking at which ones were genuinely unused. As far as I can tell, all of these are unused and can be removed.

After removing each dependency I ran `npm install` and `npm run build && npm run check-types`. I've also tried building and running the extension in development mode and installing from a file.

Of these:
- I'm fairly confident that the `@types` packages are correct. If any of these were used then the `check-types` script would fail.
- It may appear that `stream` is used from `cli.ts` and other files, but that is actually the built-in node.js stream implementation. The `stream` package is for providing the same implementation inside the browser/webviews, and we aren't using it in this context.
- I can't find any references to `minimatch`, `minimist`, `stream-chain`, or `ts-protoc-gen`.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/package.json [hunk 1]
```diff
@@ -1834,7 +1826,6 @@
     "@types/unzipper": "^0.10.1",
     "@types/vscode": "^1.67.0",
     "@types/webpack": "^5.28.0",
-    "@types/webpack-env": "^1.18.0",
```
Comment (koesie10) [inline extensions/ql-vscode/package.json:1837]: I think this one should actually stay here: in the view code, we can use functions from the Webpack environment, such as `require.resolve`, `module.hot`, etc. I don't think we're currently using any of those, but we do have access to those.

**Review by koesie10 — COMMENTED** [review]
These look correct to me, but I think `@types/webpack-env` should be in `package.json` (even though it's currently unused).

File: extensions/ql-vscode/package.json (hunk 1 above)
Comment (robertbrignull) [reply extensions/ql-vscode/package.json:1837]: I'm happy to leave `@types/webpack-env` in there 👍🏼

**Review by koesie10 — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**